pkg/service/service.go

```diff
@@ -1401,3 +1405,74 @@ func isWildcardAddr(frontend lb.L3n4AddrID) bool {
 	}
 	return net.IPv4zero.Equal(frontend.IP)
 }
+
+// SyncServicesOnDeviceChange finds and adds missing load-balancing entries for
+// new devices.
+func (s *Service) SyncServicesOnDeviceChange(nodeAddressing types.NodeAddressing) {
```

Comment:
I didn't look too closely at this function, mostly I was focused on the daemon structure & how the datapath reinitialization pieces fit together (plus you have to set a timelimit somewhere). I presume someone else looked here already, but if you do want closer review then we can find someone to take a closer look or I can look again later.